primitives/xvm/src/lib.rs

```diff
@@ -0,0 +1,19 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use codec::{Decode, Encode};
+use scale_info::TypeInfo;
+
+// TODO: add RuntimeDebug? Maybe only use it when not building for runtime?
+/// Virtual machine identifier in the XVM context
+#[derive(PartialEq, Eq, Clone, Encode, Decode, TypeInfo)]
+pub enum VmId {
+    Unsupported,
```

Comment:
What’s about ‘Unsupported = n’ notation?

Reply:
I'm not sure what you mean - haven't seen that notation with enums.

I only added this type because you require `Default` to be implemented over it when implementing tuples.
Personally, I'd remove it.